scene/gui/text_edit.cpp

```diff
@@ -2284,6 +2346,44 @@ void TextEdit::gui_input(const Ref<InputEvent> &p_gui_input) {
 					last_dblclk = OS::get_singleton()->get_ticks_msec();
 					last_dblclk_pos = mb->get_position();
 				}
+
+				// Open inline color picker
+				Ref<TextParagraph> ldata = text.get_line_data(line);
+				for (int l = 0; l < ldata->get_line_count(); l++) {
+					for (Variant k : ldata->get_line_objects(l)) {
+						if (k.get_type() == Variant::Type::ARRAY) {
+							Array info = k;
+							if (info.size() == 4 &&
+									info[0].is_num() &&
+									info[1].is_num() &&
+									info[2].is_num() &&
+									info[3].get_type() == Variant::Type::COLOR) {
+								int i_start = info[0];
+								int i_end = info[1];
+								int mode = info[2];
+								Color color = info[3];
+
+								if (i_start == col) {
+									if (mode < 0 || mode >= ColorPicker::MODE_MAX) {
+										color_picker->set_using_hex_string(true);
+									} else {
+										color_picker->set_using_hex_string(false);
+										color_picker->set_color_mode((ColorPicker::ColorModeType)mode);
+									};
+									color_picker->set_pick_color(color);
+									inline_color_line = line;
+									inline_color_start = i_start;
+									inline_color_end = i_end;
+
+									Vector2 xf_mpos = mb->get_global_position();
```

Comment:
```suggestion
									Vector2 xf_mpos = mb->get_position() + get_screen_position();
```

Or it will be off if the editor window is not maximized/not on the primary screen.